02-toolbar/style.css

```diff
@@ -0,0 +1,232 @@
+    background: red; */   
+}
+
+.btn--label {
+    cursor: pointer;
+    position: relative;
+    top: 0; 
+    left: 0;
+    font-size: 1.9rem;
+    display: flex;
+}
+
+/* Icons positioning */
+
+.toolbar__btns .btn--one .icon::before, 
+.toolbar__btns .btn--two .icon::before {
+    left: 15px;
+}
+.toolbar__btns .btn--one .icon::after,
+.toolbar__btns .btn--two .icon::after {
+    left: 23px;
+}
+
+
+.toolbar__btns .btn--three .icon::before {
+    left: 7px;
+}
+.toolbar__btns .btn--three .icon::after {
+    left: 12px;
+}
+
+
+.toolbar__btns .btn--four .icon::before {
+    left: 22px;
+}
+.toolbar__btns .btn--four .icon::after {
+    left: 32px;
+}
+
+
+.input {
+    position: absolute;
+    top: 200rem;
+}
+
+.btn--text {
+    opacity: 0;
+    transition: all 500ms ease;
+}
+
+.input:checked ~ .btn--label .btn--text {
+    opacity: 1;
+    transform: translateY(-20px);
+}
+
+.input:checked ~ .btn--label {
+    color: #223997;
+}
+
+.btn--label.icon {
+    width: 8rem;
+    margin: auto;
+}
+
+
+/* TOP AND BOTTOM ICONS  */
+
+.icon::before, 
+.icon::after {
+    font-weight: 900;
+    font-family: 'Font Awesome 5 Free';
+    position: absolute;
+    top: 0; 
+    left: 0; 
+    color: #a5a3bd;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    /* vertical-align: -50%;  */
+}
+
+.icon::before {
+    font-size: 3rem;
+}
+
+.icon::after {
+    bottom: -150%;
+    font-size: .8rem;
+    transform: scale(0);
+    transition: all 300ms ease;
+    color: #223997;
+}
+
+.input:checked ~ .icon::before {
+    top: -5rem;
+    transform: translateY(-20px);
+    transition: all .2s linear;
+}
+
+.input:checked ~ .icon::after {
+    transform: scale(100%);
+}
+
+
+/*For Font Awesome rendering across browsers*/ 
+.toolbar__btns::before {
+    display: inline-block;
+    font-style: normal;
+    font-variant: normal;
+    text-rendering: auto;
+    -webkit-font-smoothing: antialiased;
+}
+
+/* icon-one */
+.btn--search.icon::before {
+    content: '\f002';
+}
+.btn--search.icon::after {
+    content: '\f111';
+}
+
+/* icon-two */
+.btn--hot.icon::before {
+    content: '\f0e7';
+}
+.btn--hot.icon::after {
+    content: '\f111';
+}
+
+/* icon-three */
+.btn--events.icon::before {
+    content: '\f073';
+}
+.btn--events.icon::after {
+    content: '\f111';
+}
+
+/* icon-four */
+.btn--settings.icon::before {
+    content: '\f1de';
+}
+.btn--settings.icon::after {
+    content: '\f111';
+}
+
+
+
+
+/* ANIMATIONS */
+
+
+.ripples {
+    position: relative;
+    display: inline-block;
+    border-radius: 50%;
+    z-index: -1;
+    top: -40px;
+    box-shadow: inset 0 0 2px 6px #a5a3bd;
+    transform: scale(0);
+    transition: box-shadow .2s linear;
+    opacity: 1;
+}
+
+.btn--one .ripples, 
+.btn--two .ripples {
+    left: -11px;
+}
+
+.btn--three .ripples {
+    left: -21px;
+}
+
+.btn--four .ripples {
+    left: -5px;
+}
+
+.input:checked ~ .ripples {
+    content: '';
+    position: absolute;
```

Comment:
I'd recommend you to keep ripples always absolute, even when not checked.

And when using `absolute` you must _always_ declare top and left. Please review other places to ensure this.
If you forget it it will cause bugs in other browsers. For example, check safari and chrome - you'll notice the animation is broken :)